### ISSUE TITLE
Unopenable collection (only in AnkiDroid)

### ISSUE THREAD

**Issue by rngingold** [ankidroid/Anki-Android#6179]
###### Reproduction Steps

1. Could not open collection
2. Tried all the options given, except for deleting collection and creating new one
3. Changed the collection file to .bak

###### Expected Result
Problem resolved

###### Actual Result
Problem remains

###### Debug info
AnkiDroid Version = 2.10

Android Version = 9

ACRA UUID = e58f46ba-9f8c-4e39-9734-c37453bbe9f5

Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison-1** [comment]
Hi @rngingold,

Ref: https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/reports-browser/user/e58f46ba-9f8c-4e39-9734-c37453bbe9f5

Could you try downloading the file to your phone and attempting to import it.

**Comment by david-allison-1** [comment]
Hi @rngingold,

I can't see any reports of that nature. Only relevant one is a failed attempt to import: https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/reports-browser/user/e58f46ba-9f8c-4e39-9734-c37453bbe9f5

Would it be possible for you to share your `collection.anki2` file from the AnkiDroid folder on your phone, either publicly (via Dropbox/Google Drive), or via gmail: davidallisongithub

Sorry for the inconvenience,

David

**Comment by david-allison-1** [comment]
Got the collection

```
2020-05-14 16:12:27.199 6745-6835/com.ichi2.anki E/CollectionTask: doInBackgroundLoadDeckCounts - error
    java.lang.NullPointerException: Attempt to read from null array
        at com.ichi2.libanki.sched.Sched.deckDueList(Sched.java:229)
        at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.java:414)
        at com.ichi2.async.CollectionTask.doInBackgroundLoadDeckCounts(CollectionTask.java:536)
        at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:247)
        at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:73)
        at android.os.AsyncTask$2.call(AsyncTask.java:334)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:246)
        at java.util.concurrent.ThreadPoolExecutor.processTask(ThreadPoolExecutor.java:1187)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:784)
2020-05-14 16:12:27.207 6745-6745/com.ichi2.anki E/DeckPicker: null result loading deck counts
```

**Comment by david-allison-1** [comment]
Okay, simply we have two decks:

```
OLD::PED::Internato PED
OLD::PED::INTERNATO PED::NEO
```

This fails due to a casing error. Will investigate how Anki handles this

**Comment by david-allison-1** [comment]
This also fails in Anki Desktop

```
---------------------------
Anki
---------------------------
Anki was unable to open your collection file. If problems persist after restarting your computer, please use the Open Backup button in the profile manager.

Debug info:
Traceback (most recent call last):
  File "aqt\main.py", line 441, in loadCollection
  File "aqt\main.py", line 467, in _loadCollection
  File "aqt\main.py", line 588, in moveToState
  File "aqt\main.py", line 595, in _deckBrowserState
  File "aqt\deckbrowser.py", line 54, in show
  File "aqt\deckbrowser.py", line 109, in _renderPage
  File "lib\site-packages\anki\schedv2.py", line 274, in deckDueTree
  File "lib\site-packages\anki\sched.py", line 167, in deckDueList
KeyError: 'OLD::PED::INTERNATO PED'

---------------------------
OK   
---------------------------
```

**Comment by david-allison-1** [comment]
@rngingold Do you use Anki Desktop at all? Or only AnkiDroid?

**Comment by mikehardy** [comment]
The second deck should have been gated against creation somewhere. Deck names are case-insensitive so this is some part of the ecosystem creating corrupt data, the healing parts of the system (database check) not fixing corrupt data, and us at least not gracefully handling it. But the data is corrupt as the root cause

**Comment by mikehardy** [comment]
I bet we silently handled it until some of the casing changes from Arthur recently

**Comment by david-allison-1** [comment]
Should be handled by checkDeckTree:

https://github.com/ankidroid/Anki-Android/blob/988e0ba1c55ab3d89aca57152c5c8ddb9c3cf775/AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java#L778-L806

**Comment by david-allison-1** [comment]
So, `_ensureParents` should be used as `name = _ensureParents`, but this isn't going to be where we caused the corruption.

**Comment by rngingold** [comment]
> @rngingold Do you use Anki Desktop at all? Or only AnkiDroid?

Yes, I do. And it is function appropriately. Curiously those are old decks and keep only in case I need some of the informations for creating new (better made) cards. I can try renaming or even deleting them once I get access to my desktop later today

**Comment by david-allison-1** [comment]
@rngingold Cheers!

I'm still coming up blank as to what caused this. Let me know if there's anything that you can think of that you've done recently

If you have any issues on your Desktop, give me a shout via email.

**Comment by rngingold** [comment]
> Okay, simply we have two decks:
> 
> ```
> OLD::PED::Internato PED
> OLD::PED::INTERNATO PED::NEO
> ```
> 
> This fails due to a casing error. Will investigate how Anki handles this

Although it was not causing trouble in desktop, I have just edited one of those decks name via ankiweb site and my problem with AnkiDroid has been solved! THANK YOU VERY MUCH

**Comment by david-allison-1** [comment]
This has been fixed in 2.12
